File: driver_service/src/main/java/com/internship/driver_service/dto/mapper/AbstractMapper.java
package com.internship.driver_service.dto.mapper;

import org.mapstruct.BeanMapping;
import org.mapstruct.Mapper;
import org.mapstruct.MappingTarget;
import org.mapstruct.NullValuePropertyMappingStrategy;

@Mapper
public interface AbstractMapper<D, E> {
    E handleDto(D dto);

    D handleEntity(E entity);

    @BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
    void updateEntity(D dto, @MappingTarget E entity);
}

File: finance_service/src/main/kotlin/com/internship/finance_service/dto/mapper/AbstractMapper.kt
package com.internship.finance_service.dto.mapper

import com.internship.finance_service.entity.Card
import org.mapstruct.BeanMapping
import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.NullValuePropertyMappingStrategy

@Mapper(componentModel = "spring")
interface AbstractMapper<E,D> {
    fun toDto(entity: E): D
    @Mapping(target = Card.ID, ignore = true)
    fun toEntity(dto: D): E
    @BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
    fun updateEntity(user: E, dto: D)
}